File: back/api/Models/Question/ProductQuestion.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models.Question;

public class ProductQuestion
{
  [Key] public Guid ProductQuestionId { get; init; }
  [Required] public int ProductId { get; init; }
  [Required] public string QuestionersNickname { get; init; }
  [Required] public string Title { get; init; }
  [Required] public string Content { get; init; }
  [Required] public DateTimeOffset CreatedAt { get; init; }
  [Required] public bool IsApproved { get; set; }

  public virtual List<ProductQuestionAnswer> Answers { get; init; }
}

File: apps/frontend/types/ProductQuestion.ts
import { Product } from "./Product";

export type ProductQuestionAnswer = {
  questionAnswerId: string;
  questionId: string;
  answerersNickname: string;
  content: string;
  createdAt: string;
  byEmployee: boolean;
};

export type ProductQuestion = {
  questionId: string;
  productId: string;
  questionersNickname: string;
  title: string;
  content: string;
  createdAt: string;
  answers: ProductQuestionAnswer[];
};

export type ProductQuestionWithProduct = ProductQuestion & {
  product: Product;
};

export type AddProductQuestionRequestBody = Omit<
  ProductQuestion,
  "questionId" | "reviewId" | "byEmployee" | "createdAt" | "answers" | "productId"
>;
